requirements.txt

```diff
@@ -1,58 +1,58 @@
-alembic==0.9.9
-amqp==2.3.2
-attrs==18.1.0
-Babel==2.6.0
-billiard==3.5.0.3
-blinker==1.4
-celery==4.2.0
-certifi==2018.4.16
-chardet==3.0.4
-click==6.7
-click-log==0.3.2
-enum34==1.1.6
-Flask==1.0.2
-Flask-Alembic==2.0.1
-flask-allows==0.6.0
-Flask-BabelPlus==2.1.1
-Flask-Caching==1.4.0
-Flask-DebugToolbar==0.10.1
-flask-debugtoolbar-warnings==0.1.0
-Flask-Limiter==1.0.1
-Flask-Login==0.4.1
-Flask-Mail==0.9.1
-Flask-Redis==0.3.0
-Flask-SQLAlchemy==2.3.2
-Flask-Themes2==0.1.4
-flask-whooshee==0.6.0
-Flask-WTF==0.14.2
-flaskbb-plugin-conversations==1.0.3
-flaskbb-plugin-portal==1.1.1
-idna==2.7
-itsdangerous==0.24
-Jinja2==2.10
-kombu==4.2.1
-limits==1.3
-Mako==1.0.7
-MarkupSafe==1.0
-mistune==0.8.3
-olefile==0.45.1
-Pillow==5.1.0
-pluggy==0.6.0
-Pygments==2.2.0
-python-dateutil==2.7.3
-python-editor==1.0.3
-pytz==2018.4
-redis==2.10.6
-requests==2.19.1
-simplejson==3.15.0
-six==1.11.0
-speaklater==1.3
-SQLAlchemy==1.2.8
-SQLAlchemy-Utils==0.33.3
-Unidecode==1.0.22
-urllib3==1.23
-vine==1.1.4
-Werkzeug==0.14.1
-Whoosh==2.7.4
-WTForms==2.2.1
+alembic>=0.9.9
```

Comment:
Why did we unpin all deps? I thought that the requirements.txt files are using for the purpose of achieving repeatable installations of a complete environment?

Reply:
That's a good point. We could have two requirements files, one that's pinned for repeated tests with known versions. And one that runs against what's actually installed when the user installs.

Doubles the number of tests though.

Reply:
We haven't pinned the dependencies in the setup.py file. I would keep it the way we have it now -- the pinned deps in the requirements file and the "unpinned" (we did set a minimum version) in the setup.py file. It's a kinda tricky topic though..

Reply:
That's kind of what I'm getting at though. We test against one strictly defined set of dependencies but users run against a very loosely defined set of dependencies -- same libraries but different versions. If we run tests against the unpinned versions, we can see something break ahead of an issue being filed (in theory at least).

I'm fine with testing against our strictly defined ones, but we should at least one set of tests against the looser defined one. That could be just one tox task that runs this unpinned version of requirements rather than all of them doing that.